lib/driver.js

```diff
@@ -233,13 +234,14 @@ class AndroidDriver extends BaseDriver {
     // set up the device to run on (real or emulator, etc)
     this.defaultIME = await helpers.initDevice(this.adb, this.opts);
 
-    // set actual device name, udid, platform version, screen size, model and manufacturer details
+    // set actual device name, udid, platform version, screen size, model, manufacturer details and device API level.
     this.caps.deviceName = this.adb.curDeviceId;
     this.caps.deviceUDID = this.opts.udid;
     this.caps.platformVersion = await this.adb.getPlatformVersion();
     this.caps.deviceScreenSize = await this.adb.getScreenSize();
     this.caps.deviceModel = await this.adb.getModel();
     this.caps.deviceManufacturer = await this.adb.getManufacturer();
+    this.caps.deviceApiLevel = await this.adb.getApiLevel();
```

Comment:
It is safe to call `getApiLevel` without assigning it to a cap, since the value there is cached per session and does not invoke extra command line calls.

Also, I'd say the caps should be in sync in both uia1 and uia2 drivers.